### ISSUE TITLE
[MD022] consecutive increasing headers should not need to be separated by blank lines

### ISSUE THREAD

**Issue by verhovsky** [markdownlint/markdownlint#500]
This 

```md
# Header
## Subheader
### Subsubheader
```

Should be allowed. Currently the linter wants you to separate them with newlines

```md
# Header

## Subheader

### Subsubheader
```

**Comment by DavidAnson** [comment]
Spaces around headings is a correctness issue: https://github.com/DavidAnson/markdownlint/blob/main/doc/md022.md

But you are welcome to disable MD022 if it's not relevant to your scenario.

EDIT: Sorry, I was answering for the Node implementation! That said, the same logic should apply here.
